packages/mui-material/src/styles/excludeVariablesFromRoot.test.ts

```diff
@@ -0,0 +1,24 @@
+import { expect } from 'chai';
+import excludeVariablesFromRoot from './excludeVariablesFromRoot';
+
+describe('excludeVariablesFromRoot', () => {
```

Comment:
Could we add test in the `CssVarsProvider` about this logic? Also, how do we make sure we remove the variables only in light mode? Adding tests for both cases would be great.

I would argue that the alternative approach is easier to learn and apply rather than adding new API, and at this moment bundle size doesn't really concerns me that much.

Reply:
> Could we add test in the CssVarsProvider about this logic? Also, how do we make sure we remove the variables only in light mode? Adding tests for both cases would be great.

Added 2 regression tests
- Test the default dark mode to ensure that the variables exist in dark mode.
- The default mode is dark and then switches to light to ensure that those variables does not exist in light mode.

> I would argue that the alternative approach is easier to learn and apply rather than adding new API, and at this moment bundle size doesn't really concerns me that much.

I might create confusion for developers and be prone to errors:
- Introducing the `unknown` variables (do we need to discuss the name?)
- Need to document the `unknown` variables
- Developer could accidentally remove the variables in the light mode which will cause unexpected result.
- Bundle size will grow if you have more than `light` and `dark.